### PR TITLE
 task/TEC-64/business profile

### PR DESCRIPTION
**Pull request by Benrobo**
- [x] Make Category and Cities searchable both in the business creation profile and the Discover business filter. 
- [x] Selected categories and cities should be pre-filled in the placeholder.  
- [x] Selected work days should be pre-filled on the placeholder.  
- [x] Change all “Add your business“ to create a business profile.

### CODE REVIEW

File: client/src/components/Input/Select.tsx [hunk 1]
```diff
@@ -23,6 +23,12 @@ interface IOption {
 const Select = ({ label, options, name, formikValue, formik }: ISelect) => {
   //Based on the values(string) find the corresponding option (object)
   const dropdownRef = useRef<HTMLDivElement>(null);
+  // const [searchParams, setSearchParams] = useState({
+  //   businessCategory: "",
```
Comment (osxar) [inline client/src/components/Input/Select.tsx:27]: Remove commented code, if not in use

File: client/src/components/Input/Select.tsx [hunk 2]
```diff
@@ -71,7 +77,10 @@ const Select = ({ label, options, name, formikValue, formik }: ISelect) => {
             placeholder={label}
             value={value}
             onClick={toggleDropdown}
-            readOnly
+            onChange={(e) => {
+              setValue(e.target.value);
+            }}
+            // readOnly
```
Comment (osxar) [inline client/src/components/Input/Select.tsx:83]: same here

Comment (Benrobo) [reply client/src/components/Input/Select.tsx]: Done

File: client/src/components/Input/Select.tsx (hunk 1 above)
Comment (Benrobo) [reply client/src/components/Input/Select.tsx]: Done

File: client/src/components/Input/Select.tsx [hunk 3]
```diff
@@ -100,17 +109,33 @@ const Select = ({ label, options, name, formikValue, formik }: ISelect) => {
...
-                  key={option.uuid}
-                  onClick={() => handleSelect(option)}
-                  style={
-                    selectedValue?.uuid == option.uuid ? selectedStyle : {}
-                  }
-                >
-                  {option.value}
-                </li>
-              ))}
+              {value.length > 0
```
Comment (osxar) [inline client/src/components/Input/Select.tsx:105]: If we simplify this logic to this, will the functionality still hold.

```typescript
<div className="options-list">
          <ul>
            {options?.map((option) => {
              const isSelected = selectedValue?.uuid === option.uuid;
              const isMatchingValue = value.length > 0 && option.value.toLowerCase().includes(value.toLowerCase());

              if (value.length === 0 || isMatchingValue) {
                return (
                  <li
                    key={option.uuid}
                    onClick={() => handleSelect(option)}
                    style={isSelected ? selectedStyle : {}}
                  >
                    {option.value}
                  </li>
                );
              }

              return null;
            })}
          </ul>
        </div>
```

**Comment by Benrobo** [comment]
I made a mistake by including the wrong ticket number. it should be **TEC-64** and not **TEC-62**

**Comment by osxar** [comment]
As discussed, here are the updates to be made.
1) Include Search capabilities for the category tab in discover businesses page
2) Selected items in days of operations and category (i.e in discover business), should be formatted according to the figma design

**Review by osxar — APPROVED** [review]
LGTM

**Pull request merged**